### PR TITLE
ci: add linting for yml files

### PR DESCRIPTION
**Pull request by dhth**

### CODE REVIEW

**Comment by ccoVeille** [comment]
I'm surprised. Usually yamlfmt is not used for limiting. You will find quite GitHub action for that, it may explain why you had to do the trick to install the latest version

Usually, yamlint is the one used for linting.

Any reason for that?

File: .github/workflows/lint-yml.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  lint-yml:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get yamlfmt
+        run: |
+          LATEST_VERSION=$(curl -s https://api.github.com/repos/google/yamlfmt/releases/latest | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/' | sed 's/^v//')
+          ./.github/scripts/get-yamlfmt.sh "Linux" "x86_64" "$LATEST_VERSION"
```
Comment (ccoVeille) [inline .github/workflows/lint-yml.yml:20]: You could have used

`go install github.com/google/yamlfmt/cmd/yamlfmt@latest`

It would have been simpler

**Comment by dhth** [comment]
> I'm surprised. Usually yamlfmt is not used for limiting. You will find quite GitHub action for that, it may explain why you had to do the trick to install the latest version
> 
> Usually, yamlint is the one used for linting.
> 
> Any reason for that?

Well, you would use the same tool for formatting and linting, no? Between the choice between yamlfmt and yamllint, I went with the former because:

a. It's written in go, so easier to install locally, and should (theoretically) be faster
b. It's under the @google org (not that it matters too much), so hopefully will have more resources behind it (that's of course a guess on my side)

File: .github/workflows/lint-yml.yml (hunk 1 above)
Comment (ccoVeille) [reply .github/workflows/lint-yml.yml:20]: I looked for yamlfmt action and it's very disappointing

https://github.com/yk-lab/yamlfmt-action

Or this disappointing one

https://github.com/credfeto/action-yaml-format

The reformat is a sed 😂


I found this one, but yes it use setup-go then install it
https://github.com/norwd/fmtya

Comment (dhth) [reply .github/workflows/lint-yml.yml:20]: Haha, yeah, it's slim pickings for this tool.

I did find https://github.com/supplypike/setup-bin which does work for simply downloading the binary files and putting the in `$PATH`, but then I lose checksum and signing (if applicable) validation.

Comment (dhth) [reply .github/workflows/lint-yml.yml:20]: [And https://github.com/jaxxstorm/action-install-gh-release](https://github.com/jaxxstorm/action-install-gh-release) works well too

Comment (dhth) [reply .github/workflows/lint-yml.yml:20]: I guess there is no silver bullet for this use case as the validation steps can be quite varied across repositories.

**Pull request merged**